### PR TITLE
BaSD-MS 41: Create Postulants View

### PR DESCRIPTION
**Pull request by NicoCagnina**
# Create Postulants front
Completed the postulant CRUD along with all its functionalities.

## Components used:

- Modal
- List
- Postulant
- Input
- Form

### CODE REVIEW

File: src/Components/Postulants/List/index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const openNewForm = () => {
+    window.location.href = `/postulants/form`;
+  };
+  const handleDelete = (id) => {
+    const url = `${process.env.REACT_APP_API}/postulants/${id}`;
+    fetch(url, {
+      method: 'DELETE'
+    })
+      .then((res) => {
+        if (res.status !== 204) {
```
Comment (ravegliamartin) [inline src/Components/Postulants/List/index.js:18]: The res.status is 200, that's what's defined in the server-side postulant controller. Here you're catching the error and doing nothing with it, so nothing's broken. But ideally, i think this piece of code should be,

```
const handleDelete = (id) => {
    const url = `${process.env.REACT_APP_API}/postulants/${id}`;
    fetch(url, {
      method: 'DELETE'
    })
      .then((res) => {
        if (res.status !== 200) {
            return res.json().then((message) => {
            throw new Error(message);
          });
        }
      setPostulants(dataList.filter((postulants) => postulants._id !== id));
      })
      .catch((error) => {show the error somewhere});
```

File: src/Components/Postulants/List/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          return res.json().then((message) => {
+            throw new Error(message);
+          });
+        }
+      })
+      .catch((error) => error);
+    setPostulants(dataList.filter((postulants) => postulants._id !== id));
+    // };
```
Comment (ravegliamartin) [inline src/Components/Postulants/List/index.js:26]: Comments should be removed

File: src/Components/Postulants/Postulant/index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import DeleteBtn from '../DeleteBtn';
+import styles from './postulant.module.css';
+
+const Postulant = ({ object, onClick }) => {
+  const openEditForm = () => {
+    console.log(object._id);
```
Comment (ravegliamartin) [inline src/Components/Postulants/Postulant/index.js:6]: `console.log` should be removed

**Review by ravegliamartin — CHANGES_REQUESTED** [review]

File: src/Components/Postulants/Form/index.js [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+  const [countryValue, setCountryValue] = useState('');
+  const [telephoneValue, setTelephoneValue] = useState('');
+  const [jobPositionValue, setJobPositionValue] = useState('');
+  const [employerValue, setEmployerValue] = useState('');
+  const [startDateValue, setStartDateValue] = useState('');
+  const [endDateValue, setEndDateValue] = useState('');
+  const [descriptionValue, setDescriptionValue] = useState('');
+
+  const setInputValues = (data) => {
+    console.log(data);
```
Comment (nicolasviola) [inline src/Components/Postulants/Form/index.js:26]: Elimina este console.log

File: src/Components/Postulants/Form/index.js [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+            throw new Error(message);
+          });
+        }
+        return response.json();
+      })
+      .then(() => {
+        window.location.href = `/postulants`;
+      })
+      .catch((error) => {
+        console.log(error);
```
Comment (nicolasviola) [inline src/Components/Postulants/Form/index.js:113]: Elimina este console.log y crea un state para completar en caso de que haya un error y mostrarlo en el componente

File: src/Components/Postulants/Form/index.js [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+        />
+        <Input
+          name="telephoneValue"
+          value={telephoneValue}
+          onChange={(e) => setTelephoneValue(e.target.value)}
+          placeholder="Telephone"
+          required
+        />
+        <h2>Postulant Experience</h2>
+
```
Comment (nicolasviola) [inline src/Components/Postulants/Form/index.js:217]: Elimina esa linea

File: src/Components/Postulants/List/index.js (hunk 1 above)
Comment (nicolasviola) [reply src/Components/Postulants/List/index.js:18]: Yeap

**Review by nicolasviola — CHANGES_REQUESTED** [review]

**Pull request merged**